### PR TITLE
API.md: Mention mhd_response_copy_data when using custom allocator

### PR DESCRIPTION
**Pull request by laf0rge**
... also expand a bit on the slightly cryptic comment on liborcania.

### CODE REVIEW

File: API.md [hunk 1]
```diff
@@ -82,7 +82,9 @@ void o_set_alloc_funcs(o_malloc_t malloc_fn, o_realloc_t realloc_fn, o_free_t fr
 void o_get_alloc_funcs(o_malloc_t * malloc_fn, o_realloc_t * realloc_fn, o_free_t * free_fn);
 ```
 
-To link your program with Orcania library, add `-lorcania` to your linking command.
+Accessing those functions requires you to directly link your application also against the Orcania library. To do so, add `-lorcania` to your linking command.
+
+Furthermore, you need to set the `mhd_response_copy_data = 1` in your `_u_instance`, as otherwise libmicrohttpd will attempt to call normal system `free()` on memory allocated by your user-provided allocator.
```
Comment (babelouest) [inline API.md]: This sentence says that if the app using ulfius uses a different `free()` function, you must set the option `mhd_response_copy_data = 1`, but it's not mandatory. i.e. if I use a `free` function like:

```C
void my_free(void * ptr) {
  printf("free pointer %p\n", ptr);
  free(ptr);
}
```

Then there wouldn't be problems.
Although I totally understand your change request for the API documentation, but I'd rephrase like that:

```
Furthermore, you may need to set the `mhd_response_copy_data = 1` in your `_u_instance` if you use an `free()` function incompatible with the libc `free()` function.
```

What do you think?

Comment (laf0rge) [reply API.md]: I disagree, with your first point, sorry.  Even your "my_free()" function causes the "problem" that for all direct calls to free(), there is no printf statement.  So as a developer, you would assume that there is a memory leak as you get a printf() from your my_malloc() function, but none from free().  That will likely cause a lot of time wasted trying to understand what's going on.

For the second point: If you want to rephrase it the way, I would suggest to word it like "... if you use a memory allocator whose allocated return values may not directly be passed to free() or if you want to make sure all free() will always go via your user-provided free callback.", which I think is a more precise description of the requirements.

Comment (babelouest) [reply API.md]: On second thoughts, I have to say I disagree with myself too, although there wouldn't be any coomputational problems in my example, there would be a design problem indeed. Must be the heat that made me not viewing the issue...

Although I maintain that rephrasing would made my intent on the `mhd_response_copy_data` option more accurate. I like your rephrasing. If you can commit this change, I'll merge the PR.

Thanks @laf0rge !
